aten/src/ATen/native/GatedLinearUnit.cpp

```diff
@@ -3,38 +3,44 @@
 #include <ATen/native/Activation.h>
 
 namespace at {
-namespace native {
-
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-DEFINE_DISPATCH(glu_stub);
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-DEFINE_DISPATCH(glu_backward_stub);
 
-Tensor& glu_out(const Tensor& self, int64_t dim, Tensor &result) {
+namespace meta {
+TORCH_META_FUNC(glu) (
+    const Tensor& self, int64_t dim
+) {
   // this can't pass anyway because a 0-dimensional tensor has "size" 1, which
   // can't be evenly halved, but give a nicer error message here.
   TORCH_CHECK(self.dim() > 0, "glu does not support 0-dimensional tensors");
   auto wrap_dim = maybe_wrap_dim(dim, self.dim());
   const int64_t nIn = self.size(wrap_dim);
   TORCH_CHECK(nIn % 2 == 0, "Halving dimension must be even, but dimension ",
               wrap_dim, " is size ", nIn);
+
   // size output to half of input
   const int64_t selfSize = nIn / 2;
   auto newSizes = self.sizes().vec();
   newSizes[wrap_dim] = selfSize;
-  result.resize_(newSizes);
+  set_output(0, newSizes, {}, self.options(), {});
```

Comment:
I'm not sure if we should have `build_unary_op(maybe_get_output(), self);` here after setting output. I had it here before but since we don't have `structured_inherits: TensorIteratorBase` in `native_functions.yaml` I had to remove it. Note that `TensorIteratorBase` can't be used (I think?) since `DECLARE_DISPATCH(activation_fn, glu_stub);` points to `using activation_fn = void (*)(TensorIterator&);`

Reply:
also not sure if this is the right call to `set_output`; i see the 3 arg call in some places but I couldn't get type coercion/overloading to work for some reason when writing 

`set_output(0, newSizes, self.options())`

i.e. `newSizes` wasn't being coerced (cast?) to `IntArrayRef`(again that's my hypothesis for why 3 arg call didn't type check).